<launch>
    <arg name="use_sim_time" default="false" />

    <arg name="input_path" default="true" />
    <node name="input_path_marker" pkg="central_nav" exec="path_marker_node" if="$(var input_path)">
        <remap from="path" to="/robot_paths" />
        <remap from="marker" to="/raw_path_markers" />
        <param name="thickness" value="0.01" />
        <param name="frame_id" value="map" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <arg name="robots" default="true" />
    <node name="robot_marker" pkg="central_nav" exec="robot_marker_node" if="$(var robots)">
        <remap from="robot_pass" to="/robot_pass" />
        <remap from="robot_stop" to="/robot_stop" />
        <remap from="robot_poses" to="/robot_poses" />
        <remap from="marker" to="/robot_markers" />
        <param name="use_sim_time" value="$(var use_sim_time)" />
    </node>

    <arg name="rviz" default="true" />
    <arg name="map" default="" />

    <include file="$(find-pkg-share central_nav)/launch/rviz_launch.xml" if="$(var rviz)">
        <arg name="map" value="$(var map)" />
        <arg name="use_sim_time" value="$(var use_sim_time)" />
    </include>
</launch>